packages/mobile/src/screens/sign-on-screen/screens/SelectGenreScreen.tsx

```diff
@@ -1,98 +1,104 @@
 import { useCallback } from 'react'
 
-import { GENRES, convertGenreLabelToValue } from '@audius/common'
-import { setField } from 'common/store/pages/signon/actions'
 import {
-  getHandleField,
-  getNameField
-} from 'common/store/pages/signon/selectors'
-import { Formik } from 'formik'
-import { Pressable, ScrollView, View } from 'react-native'
-import { useDispatch, useSelector } from 'react-redux'
+  GENRES,
+  convertGenreLabelToValue,
+  selectGenresPageMessages as messages,
+  selectGenresSchema
+} from '@audius/common'
+import { setField } from 'common/store/pages/signon/actions'
+import { Formik, useFormikContext } from 'formik'
+import { ScrollView, View } from 'react-native'
+import { useDispatch } from 'react-redux'
+import { toFormikValidationSchema } from 'zod-formik-adapter'
 
-import { Button, Text } from 'app/components/core'
+import { Flex, SelectablePill } from '@audius/harmony-native'
 import { useNavigation } from 'app/hooks/useNavigation'
 
+import { ReadOnlyAccountHeader } from '../components/AccountHeader'
+import { Heading, Page, PageFooter } from '../components/layout'
 import type { SignUpScreenParamList } from '../types'
 
-const messages = {
-  header: 'Select Your Genres',
-  description: 'Start by picking some of your favorite genres.',
-  continue: 'Continue'
-}
-
 const genres = GENRES.map((genre) => ({
   value: genre,
   label: convertGenreLabelToValue(genre)
 }))
 
-type SelectGenreValues = Record<string, boolean>
+type SelectGenreValues = { genres: typeof GENRES }
+
+const initialValues = { genres: [] }
 
-const initialValues = genres.reduce(
-  (acc, genre) => ({
-    ...acc,
-    [genre.value]: false
-  }),
-  {} as SelectGenreValues
-)
+const ClickableGenres = () => {
+  const { values, setValues } = useFormikContext<SelectGenreValues>()
+
+  const handleChange = (value) => {
+    const newValues = { genres: [...values.genres] }
+    const valueIndex = values.genres.indexOf(value)
+
+    if (valueIndex > -1) {
+      // Already checked
+      newValues.genres.splice(valueIndex, 1)
+    } else {
+      // Unchecked
+      newValues.genres.push(value)
+    }
+    setValues(newValues)
+  }
+
+  return (
+    <ScrollView testID='genreScrollView'>
+      <Flex gap='s' direction='row' wrap='wrap'>
+        {genres.map((genre) => (
+          <SelectablePill
+            label={genre.label}
+            key={genre.value}
+            onPress={() => handleChange(genre.value)}
+            isSelected={values.genres.includes(genre.value)}
```

Comment:
seeing some performance issues and I'm wondering if this is the culprit but I have more digging to do